### DISCUSSION TITLE
Feature Request: Conversion from file.sqlite.db to file.duckdb

### DISCUSSION THREAD

**Discussion by yaminivibha** [duckdb/duckdb#8257] [Ideas]
There have been many situations where I'd like to convert from a sqlite file to a duckdb file; however, I cannot figure out the native method. at the moment, I am using a sort of scotch-taped workaround wherein I read sqlite tables into a csv one by one, and then port them into a duckdb file one by one[ (see this script). ](https://github.com/yaminivibha/duckdb_sqlite_conversions/blob/main/sqlite2duckdb.py) however, this comes with a *lot* of issues -- i spent about 5 hours figuring out types and null casting issues.

[this issue](https://github.com/duckdb/duckdb/discussions/6842) around this topic seems to imply you can attach two sources and do stuff with them both simultaneously. how can you use this function to create a .duckdb file from a sqlite database?

**Comment by bladko-tom** [comment]
Why not attach SQLite DB and than CREATE TABLE X AS SELECT * FROM SQLITE_DB.TABLE ?

**Comment by l1t1** [comment]
you need to do above in a loop to convert all tables
is there a way to find out all table names in an attached sqlite db?

**Reply by orzom411** [reply]
I think you can do this:

```
LOAD sqlite_scanner;
SELECT s.*
  FROM sqlite_scan('sqlite.db', 'sqlite_master' ) as s
 WHERE s.type='table';
```

_(sorry I didn't test this, just off the top of my head)._
